controllers/storagecluster/storageclasses.go

```diff
@@ -232,6 +234,16 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 					r.Log.Info("Failed to create StorageClass.", "StorageClass", klog.KRef(sc.Namespace, sc.Name))
 					return err
 				}
+				scRecreated = true
+			}
+			if !scRecreated {
+				util.AddAnnotation(existing, keyRotationEnableAnnotation, sc.GetAnnotations()[keyRotationEnableAnnotation])
```

Comment:
@black-dragon74 here we need to get the annotation from storagecluster and set it/Remove it right?

Reply:
The annotation will be present on `sc` (the new storageclass) only when it is present on the storagecluster. Hence here we are reading from the `sc` instead of reading from the `initData` (storagecluster) again.

Reply:
what about the case when the annotation is removed from the storagecluster? i didnt see the code where we are removing the annotation from SC, Are we setting the key and empty value? If thats the case, i would suggest to remove the key or set it to true explicitly

Reply:
Yes, `util.AddAnnotation` will set the value of the annotation to an empty string. Let's change it to:

```go
krEnabled := sc.GetAnnotations()[keyRotationEnableAnnotation]
if krEnabled != "false" {
	krEnabled = "true"
}
util.AddAnnotation(existing, keyRotationEnableAnnotation, krEnabled)
```
This is more explicit and won't set an empty value.

Reply:
Done!